src/components/DateRangeInput/DateRangeInput.tsx

```diff
@@ -230,7 +230,11 @@ const DateRangeInput: React.FC<
     if (areDatesUndefined(previousDateRange) && areDatesUndefined(currentDateRange)) {
       return true;
     }
-    if (currentDateRange[0] && !currentDateRange[1]) {
+    if (
+      currentDateRange[0] &&
```

Comment:
This might seem a bit off. It took me a while to realize why we check if the currentDateRange[0] must have a value. This is simply because we have the "Clear dates" button where a user can click to reset the dates and then he can click on the save button to reset the component values.